### PR TITLE
jemalloc: Use AutotoolsPackage and allow for arbitrary public API prefixes

### PR DESCRIPTION
**Pull request by calccrypto**
Removed `je` variant

### CODE REVIEW

File: var/spack/repos/builtin/packages/jemalloc/package.py [hunk 1]
```diff
@@ -24,22 +24,24 @@ class Jemalloc(Package):
 
     variant('stats', default=False, description='Enable heap statistics')
     variant('prof', default=False, description='Enable heap profiling')
-    variant('je', default=False, description='Prepend the public API functions with "je_"')
+    variant(
+        'jemalloc_prefix', default='none',
+        description='prefix',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jemalloc/package.py]: Could we make this description a little more descriptive? Something like:
```suggestion
        description='prefix to prepend to all public APIs',
```

File: var/spack/repos/builtin/packages/jemalloc/package.py [hunk 2]
```diff
@@ -24,22 +24,24 @@ class Jemalloc(Package):
...
-            configure_args.append('--with-jemalloc-prefix=je_')
+            args.append('--enable-prof')
 
-        configure(*configure_args)
+        je_prefix = spec.variants['jemalloc_prefix'].value
+        if je_prefix != 'none':
+            args.append('--with-jemalloc-prefix={0}'.format(je_prefix))
 
-        # Don't use -Werror
-        filter_file(r'-Werror=\S*', '', 'Makefile')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jemalloc/package.py:42]: Is this patch still needed? If so we can add it in a separate `patch` function.

File: var/spack/repos/builtin/packages/jemalloc/package.py (hunk 1 above)
Comment (calccrypto) [reply var/spack/repos/builtin/packages/jemalloc/package.py]: Whoops. That was a placeholder string that should have been changed before I pushed.

Fixed with 79a917f9f61717c27728dd04baff00ea5d1286e7

File: var/spack/repos/builtin/packages/jemalloc/package.py (hunk 2 above)
Comment (calccrypto) [reply var/spack/repos/builtin/packages/jemalloc/package.py:42]: I don't believe it is needed any more.

**Comment by calccrypto** [comment]
Branch name was wrong, so moved to #18680

**Pull request closed without merging**